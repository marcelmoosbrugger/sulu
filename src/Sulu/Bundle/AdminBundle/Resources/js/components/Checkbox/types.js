// @flow
import type {Node} from 'react';

export type CheckboxProps = {
    checked: boolean,
    value?: string | number,
    name?: string,
    onChange?: (checked: boolean, value?: string | number) => void,
    children?: Node,
};
