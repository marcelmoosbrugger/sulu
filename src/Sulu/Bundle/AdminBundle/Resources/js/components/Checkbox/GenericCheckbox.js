// @flow
import React from 'react';
import classNames from 'classnames';
import Icon from '../Icon';
import type {CheckboxProps} from './types';
import genericCheckboxStyles from './genericCheckbox.scss';

type Props = CheckboxProps & {
    className: string,
    icon?: string,
};

export default class GenericCheckbox extends React.PureComponent<Props> {
    static defaultProps = {
        className: '',
    };

    handleChange = (event: SyntheticEvent<HTMLInputElement>) => {
        if (this.props.onChange) {
            this.props.onChange(event.currentTarget.checked, this.props.value);
        }
    };

    render() {
        const checkboxClass = classNames({
            [genericCheckboxStyles.checkbox]: true,
            [this.props.className]: !!this.props.className,
        });

        return (
            <label className={genericCheckboxStyles.genericCheckbox}>
                <span className={checkboxClass}>
                    <input
                        type="checkbox"
                        name={this.props.name}
                        checked={this.props.checked}
                        onChange={this.handleChange} />
                    <span>
                        {this.props.icon && <Icon name={this.props.icon} />}
                    </span>
                </span>
                {this.props.children && <span>{this.props.children}</span>}
            </label>
        );
    }
}
