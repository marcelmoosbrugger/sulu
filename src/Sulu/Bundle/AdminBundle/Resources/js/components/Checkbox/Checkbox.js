// @flow
import React from 'react';
import classNames from 'classnames';
import type {CheckboxProps} from './types';
import checkboxStyles from './checkbox.scss';
import GenericCheckbox from './GenericCheckbox';

type Props = CheckboxProps & {
    skin: 'dark' | 'light',
};

const CHECKED_ICON = 'check';

export default class Checkbox extends React.PureComponent<Props> {
    static defaultProps = {
        skin: 'dark',
    };

    render() {
        const checkboxClass = classNames({
            [checkboxStyles.checkbox]: true,
            [checkboxStyles[this.props.skin]]: true,
        });

        return (
            <GenericCheckbox
                className={checkboxClass}
                checked={this.props.checked}
                value={this.props.value}
                name={this.props.name}
                icon={this.props.checked ? CHECKED_ICON : undefined}
                onChange={this.props.onChange}>
                {this.props.children}
            </GenericCheckbox>
        );
    }
}
