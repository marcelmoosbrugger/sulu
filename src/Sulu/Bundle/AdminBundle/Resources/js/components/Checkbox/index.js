// @flow
import GenericCheckbox from './GenericCheckbox';
import Checkbox from './Checkbox';

export default Checkbox;
export {GenericCheckbox, Checkbox};
