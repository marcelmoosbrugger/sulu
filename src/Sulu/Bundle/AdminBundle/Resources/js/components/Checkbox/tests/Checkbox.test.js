/* eslint-disable flowtype/require-valid-file-annotation */
import {shallow, render} from 'enzyme';
import React from 'react';
import Checkbox from '../Checkbox';

test('The component should render in light skin', () => {
    const checkbox = render(<Checkbox skin="light" />);
    expect(checkbox).toMatchSnapshot();
});

test('The component should render in dark skin', () => {
    const checkbox = render(<Checkbox skin="dark" />);
    expect(checkbox).toMatchSnapshot();
});


test('The component pass the props correctly to the generic checkbox', () => {
    const onChange = () => 'my-on-change';
    const checkbox = shallow(
        <Checkbox
            onChange={onChange}
            value="my-value"
            name="my-name"
            checked={true}>My label</Checkbox>
    );
    const genericCheckbox = checkbox.find('GenericCheckbox');
    expect(genericCheckbox.props().value).toBe('my-value');
    expect(genericCheckbox.props().name).toBe('my-name');
    expect(genericCheckbox.props().checked).toBe(true);
    expect(genericCheckbox.props().children).toBe('My label');
    expect(genericCheckbox.props().onChange()).toBe('my-on-change');
});
