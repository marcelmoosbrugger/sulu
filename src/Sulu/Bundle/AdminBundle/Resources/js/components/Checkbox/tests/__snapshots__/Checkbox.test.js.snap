// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render in dark skin 1`] = `
<label
  class="genericCheckbox"
>
  <span
    class="checkbox checkbox dark"
  >
    <input
      type="checkbox"
    />
    <span />
  </span>
</label>
`;

exports[`The component should render in light skin 1`] = `
<label
  class="genericCheckbox"
>
  <span
    class="checkbox checkbox light"
  >
    <input
      type="checkbox"
    />
    <span />
  </span>
</label>
`;
